recommender/recommender.py

```diff
@@ -27,12 +27,37 @@ def emit(param1, param2):
 from mako.lookup import TemplateLookup
 
 from xblock.core import XBlock
-from xblock.fields import Scope, List, Dict
+from xblock.fields import Scope, List, Dict, Boolean, String
 from xblock.fragment import Fragment
 from xblock.reference.plugins import Filesystem
 
 from webob.response import Response
 
+
+def stem_url(url):
```

Comment:
Seems like you're better off using a genuine URL parsing library for this, like [urlobject](https://urlobject.readthedocs.org/en/latest/quickstart.html).